iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.inl

```diff
@@ -25,50 +25,64 @@ namespace iox
 {
 namespace cxx
 {
+// AXIVION Next Construct AutosarC++19_03-A12.6.1: members are initialized in body before read access
 template <uint64_t Capacity, typename ReturnType, typename... Args>
 template <typename Functor, typename>
 inline storable_function<Capacity, signature<ReturnType, Args...>>::storable_function(const Functor& functor) noexcept
 {
     storeFunctor(functor);
 }
 
+// AXIVION Next Construct AutosarC++19_03-A12.1.5: constructor delegation is not feasible here due
+// to lack of sufficient common initialization
+// AXIVION Next Construct AutosarC++19_03-A12.6.1: members are initialized in body before read access
 template <uint64_t Capacity, typename ReturnType, typename... Args>
 inline storable_function<Capacity, signature<ReturnType, Args...>>::storable_function(
     ReturnType (*function)(Args...)) noexcept
-    : /// @NOLINTJUSTIFICATION we use type erasure in combination with compile time template arguments to restore
+    : // AXIVION Next Construct AutosarC++19_03-M5.2.6: the converted pointer is only used
+      // as its original function pointer type after reconversion (type erasure)
+      // AXIVION Next Construct AutosarC++19_03-A5.2.4: reinterpret_cast is required for type erasure
+      /// @NOLINTJUSTIFICATION we use type erasure in combination with compile time template arguments to restore
       ///                      the correct type whenever the callable is used
```

Comment:
The same justification can be found for the Axivion warning one line above. So please remove this one here. For readability I would suggest to move the Axivion suppressions above the c'tor.